src/history/StateSnapshot.cpp

```diff
@@ -149,7 +149,8 @@ StateSnapshot::differingHASFiles(HistoryArchiveState const& other)
     {
         auto b = mApp.getBucketManager().getBucketByHash(hexToBin256(hash));
         releaseAssert(b);
-        addIfExists(std::make_shared<FileTransferInfo>(*b));
+        addIfExists(std::make_shared<FileTransferInfo>(
```

Comment:
FileTransferInfo I think will need to use `getPrimaryFile` or else it'll break on switch, no?